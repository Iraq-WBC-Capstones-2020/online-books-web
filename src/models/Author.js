import { authorsRef, booksRef } from './../api/firebase';
import Book from './Book';
import { authors } from './data';
export default class Author {
  constructor({ id, name, avatar, bio, socialMedia } = {}) {
    this.id = id;
    this.name = name;
    this.avatar = avatar;
    this.bio = bio;
    this.socialMedia = socialMedia ? socialMedia : [];
  }
  static async getAuthorFromDB(id) {
    const doc = await authorsRef.doc(id).get();
    return new Author(doc.data());
  }
  async getBooks() {
    if (this.books) return this.books;
    const querySnapshot = await booksRef
      .where('author_id', '==', this.id)
      .get();
    const books = querySnapshot.docs.map((book) => new Book(book.data()));
    this.books = books;
    return books;
  }
  static async getAuthors() {
    const documentData = await authorsRef.get();
    const authors = documentData.docs.map(
      (author) => new Author(author.data())
    );
    return authors;
  }
  async updateAuthorInfo() {
    if (!this.id) this.id = authorsRef.doc().id;
    await authorsRef.doc(this.id).set({ ...this });
  }
  static async sendAuthors() {
    const authorss = authors.map((author) => new Author(author));
    for (const author of authorss) {
      await author.updateAuthorInfo();
    }
  }
}
