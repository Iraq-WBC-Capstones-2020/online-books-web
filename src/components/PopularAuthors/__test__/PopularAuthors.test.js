import React from 'react';
import renderer from 'react-test-renderer';
import Author from './../PopularAuthors';

test('PopularAuthors snapshot', () => {
  const tree = renderer.create(<Author />).toJSON();
  expect(tree).toMatchSnapshot();
});
