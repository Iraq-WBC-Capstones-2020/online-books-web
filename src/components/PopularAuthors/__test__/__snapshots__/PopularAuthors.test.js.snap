// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularAuthors snapshot 1`] = `
<div>
  <div
    className="container mt-5"
    data-test="container"
  >
    <div>
      <h1
        className="text-center text-3xl"
      >
        mainPage.PopularAuthors
      </h1>
    </div>
    <div
      className="row mt-5"
      data-test="row"
    >
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Zen Cho
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          John Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Ahemd Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Ali Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Alif Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          john Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          rema bedr
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Ali Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Alif Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          john Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Ali Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="col-12 col-sm-2 col-md-3 mt-4 d-flex justify-content-center align-items-center"
        data-test="col"
      >
        <button
          className="btn-light-blue btn Ripple-parent px-2 rounded-pill button"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          John Doe
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
